src/Unused/CLI/Search.hs

```diff
@@ -0,0 +1,36 @@
+module Unused.CLI.Search
+    ( SearchRunner(..)
+    , executeSearch
+    ) where
+
+import Unused.CLI.SearchWithProgress (searchWithProgressBar)
+import Unused.CLI.SearchWithoutProgress (searchWithoutProgressBar)
+import Unused.CLI.Util
+
+data SearchRunner = SearchWithProgress | SearchWithoutProgress
+
+executeSearch :: SearchRunner -> [String] -> IO [String]
+executeSearch runner terms = do
+    resetScreen
+    printAnalysisHeader terms
+    results <- runSearch runner $ terms
```

Comment:
This `$` is redundant. If you're interested, you could remove the binding by doing

```hs
resetScreen
printAnalysisHeader terms
runSearch runner terms <* resetScreen
```

`(<*)` is kind of like, "do these two things, but return the result of the left one." (There's also `(*>)` for the opposite.)